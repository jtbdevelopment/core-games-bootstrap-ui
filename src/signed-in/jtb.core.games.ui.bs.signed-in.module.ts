import {NgModule} from '@angular/core';
import {BrowserModule} from '@angular/platform-browser';
import {NgbModule} from '@ng-bootstrap/ng-bootstrap';
import {SignedInComponent} from './signed-in.component';
import {JTBCoreGamesUI} from 'jtb-core-games-ui';

@NgModule({
    imports: [
        BrowserModule,
        NgbModule,
        JTBCoreGamesUI
    ],
    exports: [
        SignedInComponent
    ],
    declarations: [
        SignedInComponent
    ]
})
export class JTBCoreGamesUIBSSignedInModule {
}

