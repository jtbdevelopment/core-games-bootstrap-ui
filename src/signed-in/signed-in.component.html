<div class="signed-in container-fluid">
    <div class="row text-center">
        <div class="col">
            <p>Login successful...</p>
        </div>
    </div>
</div>
