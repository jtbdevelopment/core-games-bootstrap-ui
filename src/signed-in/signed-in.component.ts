import {Component} from '@angular/core';
import {Router} from '@angular/router';
import {PlayerService} from 'jtb-core-games-ui';

@Component({
    selector: 'signed-in',
    template: require('./signed-in.component.html'),
})
export class SignedInComponent {
    constructor(private playerService: PlayerService, private router: Router) {
        this.playerService.loadLoggedInPlayer();
        //noinspection JSIgnoredPromiseFromCall
        this.router.navigateByUrl('/main');
    }
}
