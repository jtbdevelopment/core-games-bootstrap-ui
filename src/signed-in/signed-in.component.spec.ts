import {TestBed} from '@angular/core/testing';
import {SignedInComponent} from './signed-in.component';
import {Router} from '@angular/router';
import {PlayerService} from 'jtb-core-games-ui';


class MockPlayerService {
  loadLoggedInPlayer = jasmine.createSpy('llip');
}

class MockRouter {
  navigateByUrl = jasmine.createSpy('nbu');
}

describe('Component:  signed in component', () => {
  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [
        SignedInComponent,
      ],
      providers: [
        {provide: PlayerService, useClass: MockPlayerService},
        {provide: Router, useClass: MockRouter}
      ],
    });
    TestBed.compileComponents();
  });

  it('tells player service to load and navigates', () => {
    const fixture = TestBed.createComponent(SignedInComponent);
    let mockPlayerService = fixture.debugElement.injector.get(PlayerService);
    let router = fixture.debugElement.injector.get(Router);
    expect(mockPlayerService.loadLoggedInPlayer).toHaveBeenCalled();
    expect(router.navigateByUrl).toHaveBeenCalledWith('/main');
  });

  it('displays login successful', () => {
    const fixture = TestBed.createComponent(SignedInComponent);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
