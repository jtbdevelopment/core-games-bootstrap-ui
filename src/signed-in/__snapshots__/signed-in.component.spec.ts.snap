// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:  signed in component displays login successful 1`] = `
<signed-in
  playerService={[Function MockPlayerService]}
  router={[Function MockRouter]}
>
  <div
    class="signed-in container-fluid"
  >
    
    
    <div
      class="row text-center"
    >
      
      
      <div
        class="col"
      >
        
        
        <p>
          Login successful...
        </p>
        
      
      </div>
      
    
    </div>
    
  
  </div>
  
</signed-in>
`;
